From 73ebc2f1e1a2944887c9aaf7b97c6af3dbc9adaf Mon Sep 17 00:00:00 2001
From: lily <floridsleeves@gmail.com>
Date: Mon, 22 Aug 2022 22:44:11 -0700
Subject: [PATCH 279/595] [Backport] net/core/skbuff: Check the return value of
 skb_copy_bits()

stable inclusion
from stable-v5.10.143
commit 9d040a629e7e7965c70e3ff16e1a8180470b8746
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5ZG5T

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=9d040a629e7e7965c70e3ff16e1a8180470b8746

--------------------------------

[ Upstream commit c624c58e08b15105662b9ab9be23d14a6b945a49 ]

skb_copy_bits() could fail, which requires a check on the return
value.

Signed-off-by: Li Zhong <floridsleeves@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 net/core/skbuff.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index bb31ac64ec38..349e47ce6ed3 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -4063,9 +4063,8 @@ struct sk_buff *skb_segment(struct sk_buff *head_skb,
 				SKB_GSO_CB(nskb)->csum_start =
 					skb_headroom(nskb) + doffset;
 			} else {
-				skb_copy_bits(head_skb, offset,
-					      skb_put(nskb, len),
-					      len);
+				if (skb_copy_bits(head_skb, offset, skb_put(nskb, len), len))
+					goto err;
 			}
 			continue;
 		}
-- 
2.17.1

